gitlab/resource_gitlab_deploy_key_enable.go

```diff
@@ -53,7 +53,8 @@ func resourceGitlabDeployEnableKey() *schema.Resource {
 				Description: "Can deploy key push to the project’s repository.",
 				Type:        schema.TypeBool,
 				Optional:    true,
-				Computed:    true,
+				Default:     false,
+				ForceNew:    true,
```

Comment:
I don't see a reason for needing to change the schema here, so we should avoid doing so.

In your Create function you can check if `can_push` is set using `d.GetOk` and then conditionally make the update API call.

You can add a `resourceGitlabDeployKeyEnableUpdate` function and check if `can_push` has changed using `d.HasChange`, and then make the update API call.

Reply:
So, not changing the `Computed: true` as well?

Reply:
We'll see that tomorrow, tired tonight.

Reply:
Yeah no rush. We're trying to do more frequent provider releases anyway. Thanks for the PR! 😄

Reply:
I think we can leave the attribute as Optional+Computed unless anyone else disagrees.